lib/fileio/utils/meshUtil.cpp

```diff
@@ -323,5 +824,315 @@ TfToken UsdMayaMeshUtil::GetSubdivFVLinearInterpolation(const MFnMesh& mesh)
+                        int tmpOppositeVertexId;
+                        statusOK = vertIt.getOppositeVertex(tmpOppositeVertexId, connectedEdges[e]);
+                        if (!statusOK)
+                            break;
+                        if ( subdCreaseIndices[creaseIndexBase+i+1] == tmpOppositeVertexId ) {
+                            edgeIndex = connectedEdges[e];
+                            break;
+                        }
+                    }
+                    if (statusOK && edgeIndex != -1) {
+                        if (USE_CREASE_SETS) {
+                            int prevIndexDummy; // dummy param
+                            statusOK = edgeIt.setIndex(edgeIndex,prevIndexDummy);
+                            if (!statusOK)
+                                break;
+                            statusOK = elemsPerWeight[subdCreaseSharpnesses[creaseGroup]].
+                                            add(meshPath,edgeIt.currentItem());
+                            if (!statusOK)
+                                break;
+                        } else {
+                            mayaCreaseEdgeIds.append(edgeIndex);
+                            mayaCreaseEdgeValues.append(subdCreaseSharpnesses[creaseGroup]);
+                        }
+                    }
+                }
+            }
+
+            if (statusOK && !USE_CREASE_SETS) {
+                statusOK = meshFn.setCreaseEdges(mayaCreaseEdgeIds, mayaCreaseEdgeValues);
+            }
+
+            if (!statusOK) {
+                TF_RUNTIME_ERROR("Unable to set Crease Edges on <%s>: %s", 
+                        meshFn.fullPathName().asChar(),
+                        statusOK.errorString().asChar());
+                return MS::kFailure;
+            }
+
+        } else {
+            TF_RUNTIME_ERROR(
+                    "Mismatch between Crease Lengths & Sharpness on <%s>", 
+                    mesh.GetPrim().GetPath().GetText());
+            return MS::kFailure;
+        }
+    }
+
+    if (USE_CREASE_SETS) {
+        TF_FOR_ALL(weightList, elemsPerWeight) {
+            double creaseLevel = weightList->first;
+            MSelectionList &elemList = weightList->second;
+
+            if (!addCreaseSet( meshFn.name().asChar(),
+                                creaseLevel, elemList, &statusOK )){
+                TF_RUNTIME_ERROR("Unable to set crease sets on <%s>: %s", 
+                        meshFn.fullPathName().asChar(),
+                        statusOK.errorString().asChar());
+                return MS::kFailure;
+            }
+        }
+    }
+
+    return MS::kSuccess;
+}
+
+MStatus 
+UsdMayaMeshUtil::setName(const MObject& meshObj, const std::string& shapeName)
+{
+    if(meshObj.apiType() != MFn::kMesh){
+        return MS::kFailure;
+    }
+
+    MStatus status;
+
+    MFnMesh meshFn(meshObj);
+    meshFn.setName(MString(shapeName.c_str()), false, &status);
+
+    return status;
+}
+
```

Comment:
Good point. Please see 7be630d

I originally stick this here because it was needed for this PR. I have a larger PR locally for cleaning up the utility functions and didn't want bring those changes here.